x/emissions/types/conversions.go

```diff
@@ -0,0 +1,307 @@
+package types
+
+import (
+	"cosmossdk.io/errors"
+)
+
+// NewInferenceFromInput converts InputInference to Inference
+func (bi *InputInference) NewInferenceFromInput() (*Inference, error) {
```

Comment:
```suggestion
func NewInferenceFromInput(bi *InputInference) (*Inference, error) {
```
Not a big deal, it's just more idiomatic to take inputs as arguments for a constructor

Reply:
applied to all functions